src/dotnet/OpenAPI.Validator/Validation/ArmResourcePropertiesBag.cs

```diff
@@ -47,19 +47,54 @@ public class ArmResourcePropertiesBag : TypedRule<Dictionary<string, Schema>>
         public override ServiceDefinitionDocumentState ValidationRuleMergeState => ServiceDefinitionDocumentState.Composed;
 
         private static readonly IEnumerable<string> ArmPropertiesBag = new List<string>()
-                                                                        { "name", "id", "type", "location", "tag" };
+                                                                        { "name", "id", "type", "location", "tags" };
 
         // Verifies whether ARM resource has the set of properties repeated in its property bag
         public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> definitions, RuleContext context)
         {
-            var resModels = context.ResourceModels;
-            var violatingModels = resModels.Where(res => definitions[res].Properties?.ContainsKey("properties") == true
-                                                         && definitions[res].Properties["properties"]?.Properties?.Keys.Intersect(ArmPropertiesBag).Any() == true);
-            foreach (var violatingModel in violatingModels)
+            IEnumerable<string> resourceModelsWithPropertiesBag = context.ResourceModels.Where(resourceModel => definitions[resourceModel].Properties?.ContainsKey("properties") == true);
+            foreach(string resourceModelName in resourceModelsWithPropertiesBag)
             {
-                var violatingProperties = definitions[violatingModel].Properties.Keys.Union(definitions[violatingModel].Properties["properties"].Properties.Keys).Intersect(ArmPropertiesBag);
-                yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty(violatingModel).AppendProperty("properties")), this, violatingModel,
-                                                   string.Join(", ", violatingProperties));
+                Dictionary<string, IEnumerable<string>> violations = new Dictionary<string, IEnumerable<string>>();             
+                CheckModelForViolation(definitions[resourceModelName], resourceModelName, definitions, context, violations);
+
+                foreach (string modelName in violations.Keys)
+                {
+                    yield return new ValidationMessage(new FileObjectPath(context.File,
+                    context.Path.AppendProperty(modelName).AppendProperty("properties")), this, modelName,
+                                                   string.Join(", ", violations[modelName]));
+                }
+            }
+        }
+
+        private string GetName(string reference)
```

Comment:
should we be using StripDefinitionPath method?: https://github.com/Azure/azure-openapi-validator/blob/42ae8eedab0a3894443966e4513fd28ae5aa77d8/src/dotnet/OpenAPI.Validator/Extensions.cs#L96